dateutil/test/test_relativedelta.py

```diff
@@ -303,6 +303,41 @@ def testComparison(self):
         self.assertEqual(d1, d2)
         self.assertNotEqual(d1, d3)
 
+    def testNotImplementedComparison(self):
+        with self.assertRaises(TypeError):
+            relativedelta(month=1) > relativedelta(days=30)
+
+        with self.assertRaises(TypeError):
+            relativedelta(month=1) >= relativedelta(days=30)
+
+        with self.assertRaises(TypeError):
+            relativedelta(month=1) < relativedelta(days=30)
+
+        with self.assertRaises(TypeError):
+            relativedelta(month=1) <= relativedelta(days=30)
+
+    def textCustomComparison(self):
+        if PY2:
+            class AlwaysLessThan:
```

Comment:
I believe you can achieve essentially the same thing using an existing object I created, [`ComparesEqual`](https://github.com/dateutil/dateutil/blob/master/dateutil/test/_common.py#L267), which compares equal to everything, so this test can be rewritten as:

```python
rd = relativedelta(month=1)
assert not rd < ComparesEqual
assert not rd > ComparesEqual
assert rd >= ComparesEqual
assert rd <= ComparesEqual
```